src/main/java/org/opensearch/knn/index/util/Lucene.java

```diff
@@ -24,6 +24,8 @@
  */
 public class Lucene extends JVMLibrary {
 
+    private boolean initialized;
```

Comment:
Should this be atomic? Also, could we set this at JVMLibrary level?

Reply:
Agree on moving it to the parent class. I'm not sure on making it Atomic, effectively we're setting it only to "true" from initial "false" value, so I'm not sure if there is a possible situation when it got updated incorrectly. This would be different for the counter field, but for boolean flag like this one I think it's not necessary.